atomic_reactor/plugins/pre_inject_yum_repos.py

```diff
@@ -0,0 +1,273 @@
+
+        if errors:
+            raise ValueError('Errors found while checking yum repo urls: \n{}'
+                             .format('\n'.join(errors)))
+
+    def _final_user_line(self):
+        user = self._find_final_user()
+        if user:
+            return user
+
+        if not self.workflow.dockerfile_images.base_from_scratch:
+            # Inspect any platform: the User should be equal for all platforms
+            inspect = self.workflow.imageutil.base_image_inspect()
+            user = inspect.get(INSPECT_CONFIG, {}).get('User')
+            if user:
+                return f'USER {user}'
+
+        return ''
+
+    def _find_final_user(self):
+        """Find the user in USER instruction in the last build stage"""
+        dockerfile = self.workflow.build_dir.any_platform.dockerfile_with_parent_env(
+            self.workflow.imageutil.base_image_inspect())
+        for insndesc in reversed(dockerfile.structure):
+            if insndesc['instruction'] == 'USER':
+                return insndesc['content']  # we will reuse the line verbatim
+            if insndesc['instruction'] == 'FROM':
+                break  # no USER specified in final stage
+
+    def _cleanup_lines(self, platform):
+        lines = [
+            "RUN rm -f " + " ".join(
+                (f"'{repo.dst_filename}'" for repo in self.yum_repos[platform])
+            )
+        ]
+        if self._builder_ca_bundle:
+            lines.append(f'RUN rm -f /tmp/{self._ca_bundle_pem}')
+
+        final_user_line = self._final_user_line()
+        if final_user_line:
+            lines.insert(0, "USER root")
+            lines.append(final_user_line)
+
+        return lines
+
+    def add_koji_repo(self):
+        xmlrpc = get_koji_session(self.workflow.conf)
+        pathinfo = self.workflow.conf.koji_path_info
+        proxy = self.workflow.conf.yum_proxy
+
+        if not self.target:
+            self.log.info('no target provided, not adding koji repo')
+            return
+
+        target_info = xmlrpc.getBuildTarget(self.target)
+        if target_info is None:
+            self.log.error("provided target '%s' doesn't exist", self.target)
+            raise RuntimeError("Provided target '%s' doesn't exist!" % self.target)
+        tag_info = xmlrpc.getTag(target_info['build_tag_name'])
+
+        if not tag_info or 'name' not in tag_info:
+            self.log.warning("No tag info was retrieved")
+            return
+
+        repo_info = xmlrpc.getRepo(tag_info['id'])
+
+        if not repo_info or 'id' not in repo_info:
+            self.log.warning("No repo info was retrieved")
+            return
+
+        # to use urljoin, we would have to append '/', so let's append everything
+        baseurl = pathinfo.repo(repo_info['id'], tag_info['name']) + "/$basearch"
+
+        self.log.info("baseurl = '%s'", baseurl)
+
+        repo = {
+            'name': 'atomic-reactor-koji-plugin-%s' % self.target,
```

Comment:
since we don't have anymore koji plugin, rename to 'atomic-reactor-koji-target-%s'

Reply:
updated